### PR TITLE
Fixes #60: Keep newlines in doc strings

### LINKED ISSUE #60: Newlines are removed from doc strings
It seems Tickspec is filtering out all empty lines in a [doc string](https://relishapp.com/cucumber/cucumber/docs/gherkin/doc-strings), e.g. 

```
...
Then the generated file has the following content
"""
line1

line3

line5

"""
```

The resulting argument passed to the Then step looks like 

```
line1
line3
line5
```

Can you pls confirm this is a bug in TickSpec (Gherkin seems to preserve newlines in doc strings) ?

### PR DESCRIPTION
**Pull request by mchaloupka**

### CODE REVIEW

File: TickSpec.Tests/FeatureParserTest.fs [hunk 1]
```diff
@@ -16,20 +16,30 @@ let private verifyParsing (lines: string[]) (expected: FeatureSource) =
...
-            | Some line -> (lastLine, line)
+            | Some line -> (lineNumber, lineContent, line) |> Line
+            | None when lineContent.Trim().Length = 0 -> lastParsedLine |> IgnoredLine
             | None ->
-                let e = expectingLine lastLine
-                Exception(e) |> raise) (FileStart, FileStart)
-        |> Seq.map (fun (_, line) -> line)
+                let e = expectingLine lastParsedLine
+                Exception(e) |> raise) (Line (0, "", FileStart))
+        |> Seq.choose (function
```
Comment (bartelink) [inline TickSpec.Tests/FeatureParserTest.fs:38]:
```suggestion
        |> Seq.filter (function IgnoredLine -> false | Line -> true)
```

File: TickSpec.Tests/FeatureParserTest.fs [hunk 2]
```diff
@@ -16,20 +16,30 @@ let private verifyParsing (lines: string[]) (expected: FeatureSource) =
...
+            if i = -1 then lineNumber, line
+            else lineNumber, line.Substring(0, i)
         )
-        |> Seq.filter (fun line -> line.Trim().Length > 0)
-        |> Seq.scan (fun (_, lastLine) line ->
-            let parsed = parseLine (lastLine, line)
+        |> Seq.scan (fun prevLine (lineNumber, lineContent) ->
+            let lastParsedLine =
+                match prevLine with
+                | Line (_, _, prevLineType) -> prevLineType
```
Comment (bartelink) [inline TickSpec.Tests/FeatureParserTest.fs:28]:
```suggestion
                | Line (_, _, prevLineType)
```

File: TickSpec.Tests/FeatureParserTest.fs [hunk 3]
```diff
@@ -16,20 +16,30 @@ let private verifyParsing (lines: string[]) (expected: FeatureSource) =
 let private verifyLineParsing (lines: string[]) (expected: LineType list) =
     let lineParsed =
         lines
-        |> Seq.map (fun line ->
+        |> Seq.mapi (fun lineNumber line -> (lineNumber + 1, line))
```
Comment (bartelink) [inline TickSpec.Tests/FeatureParserTest.fs:19]: I dont think the `mapi >> map` makes things more legible - better to collapse to a single `mapi`

File: TickSpec.Tests/FeatureParserTest.fs (hunk 1 above)
Comment (bartelink) [reply TickSpec.Tests/FeatureParserTest.fs:38]: But on reflection you could collapse the `filter >> map` into a `choose`:
` |> Seq.choose (function IgnoredLine -> None | Line (_, _, line) -> Some line`  ?

File: TickSpec/BlockParser.fs [hunk 4]
```diff
@@ -230,6 +230,12 @@ let private parseFeatureFile parsedLines =
     | (_,_,FileStart) :: xs -> parseFeatureBlock xs
     | _ -> parsedLines |> raiseParseException "Unexpected call of parser"
 
+type internal ScannedLine =
```
Comment (bartelink) [inline TickSpec/BlockParser.fs:233]: perhaps RequireQualifiedAccess and `Line.Ok` and `Line.Ignore` ?

**Review by bartelink — APPROVED** [review]
Looks good; minor suggestions included that you can take or leave

File: TickSpec/BlockParser.fs (hunk 4 above)
Comment (mchaloupka) [reply TickSpec/BlockParser.fs]: Sounds good!

File: TickSpec.Tests/FeatureParserTest.fs (hunk 3 above)
Comment (mchaloupka) [reply TickSpec.Tests/FeatureParserTest.fs]: Makes sense.

File: TickSpec.Tests/FeatureParserTest.fs (hunk 2 above)
Comment (mchaloupka) [reply TickSpec.Tests/FeatureParserTest.fs]: Good suggestion, did it also in the BlockParser.

File: TickSpec.Tests/FeatureParserTest.fs (hunk 1 above)
Comment (mchaloupka) [reply TickSpec.Tests/FeatureParserTest.fs:38]: I used the suggestion - you are right, there is no point to have `choose` followed by `map`.

**Pull request merged**